src/ApiCheck.Task/ApiCheckGenerateTask.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.BuildTools.ApiCheck.Task
+{
+    /// <summary>
+    /// An MSBuild task that acts as a shim to <c>Microsoft.AspNetCore.BuildTools.ApiCheck.exe generate ...</c> or
+    /// <c>dotnet Microsoft.AspNetCore.BuildTools.ApiCheck.dll generate ...</c>.
+    /// </summary>
+    public class ApiCheckGenerateTask : ApiCheckTasksBase
+    {
+        public ApiCheckGenerateTask()
+        {
+            // Tool does not use stderr for anything. Treat everything that appears there as an error.
+            LogStandardErrorAsError = true;
+        }
+
+        protected override bool ValidateParameters()
+        {
+            if (string.IsNullOrEmpty(ApiListingPath) || !File.Exists(ApiListingPath))
+            {
+                Log.LogError($"API listing file '{ApiListingPath}' not specified or does not exist.");
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(AssemblyPath) || !File.Exists(AssemblyPath))
```

Comment:
Remaining checks should be moved into `base.ValidateParameters()` and thereby shared with `ApiCheckTask`